console/console-server/src/main/resources/iot.schema.graphql

```diff
@@ -0,0 +1,231 @@
+}
+
+type AddressSpaceConfig_iot_enmasse_io_v1alpha1  {
+  name: String!
+  plan: String!
+  type: String!
+}
+
+type AddressConfig_iot_enmasse_io_v1alpha1  {
+  plan: String!
+  type: String!
+}
+
+type AddressesConfig_iot_enmasse_io_v1alpha1  {
+  Telemetry: AddressConfig_iot_enmasse_io_v1alpha1!
+  Event: AddressConfig_iot_enmasse_io_v1alpha1!
+  Command: AddressConfig_iot_enmasse_io_v1alpha1!
+}
+
+
+type managedDownstreamStrategy_iot_enmasse_io_v1alpha1 {
+  addressSpace: AddressSpaceConfig_iot_enmasse_io_v1alpha1!
+  addresses: AddressesConfig_iot_enmasse_io_v1alpha1!
+}
+
+type externalDownstreamStrategy_iot_enmasse_io_v1alpha1 {
+  connectionInformation: connectionInformation_iot_enmasse_io_v1alpha1!
+}
+
+type IotProjectSpec_iot_enmasse_io_v1alpha1 {
+  downstreamStrategy: iotProjectDownStreamStrategy !
+  configuration: String!
+}
+
+type IoTProject_iot_enmasse_io_v1alpha1 {
+
+  metadata: ObjectMeta_v1! @goField(name: "ObjectMeta")
+  enabled: Boolean!
+  spec: IotProjectSpec_iot_enmasse_io_v1alpha1!
+  status: IoTProjectStatus_iot_enmasse_io_v1alpha1!
+  devices(iotproject: String!, first: Int, offset: Int, filter: String, orderBy: String): devicesQueryResult_consoleapi_iot_enmasse_io_v1alpha1
+  adapters: [adapter]
+}
+
+interface adapter {}
+
+# TODO : map to k8s objectcs and pull endpoints details from openshift routes ?
+type HTTP_Adapter implements adapter {
+  url : String!
+}
+
+type MQTT_Adapter implements adapter {
+ tls_enabled: Boolean!
+ host : String!
+ port : Int!
+}
+
+type AMQP_Adapter implements adapter {
+  url : String!
+}
+
+type CoAP_Adapter implements adapter {
+  url : String!
+}
+
+type device {
+  deviceID: String!
+  enabled: Boolean!
+  viaGateway: Boolean!
+  jsonData: String! #The Json representation of this device.
+}
+
+type credential{
+  authID: String!
+  enabled: Boolean!
+  type: iotCredentials!
+  ext : String #A Json Object.
+  secrets: String! #A Json Array containing the secrets.
+}
+
+type devicesQueryResult_consoleapi_iot_enmasse_io_v1alpha1 {
+  total: Int!
+  devices: [device!]!
+}
+
+type credentialsQueryResult_consoleapi_iot_enmasse_io_v1alpha1 {
+  total: Int!
+  credentials: [credential!]!
+}
+
+type addressSpaceAndIotProjectsQueryResult_consoleapi_iot_enmasse_io_v1alpha1 {
+  total: Int!
+  addressSpaces: [AddressSpace_consoleapi_enmasse_io_v1beta1!]
+  iotProjects: [IoTProject_iot_enmasse_io_v1alpha1!]
+}
+
+enum projectQueryType {
+  iotProject
+  addressSpace
+}
+
+type Query {
+
+  "Returns the namespaces and iotprojects visible to this user, optionaly filtered by project type"
+  #This extends the existing "addressSpaces" query
+  allProjects(first: Int, offset: Int, filter: String, orderBy: String, projectType: projectQueryType): addressSpaceAndIotProjectsQueryResult_consoleapi_iot_enmasse_io_v1alpha1!
+
+  "Returns the iot project tied to the given nameSpace"
+  iotProject(namespace: String!): IoTProject_iot_enmasse_io_v1alpha1!
```

Comment:
Below information seems to missing in response of ioTProject query  for project detail page:-
https://marvelapp.com/5gbji22/screen/68240479
- Max Connection
- Data Volume
- Start date
- End date
![image](https://user-images.githubusercontent.com/29524461/79994531-bc47a800-84d3-11ea-9737-fc4045f772a3.png)
- Information of Registration API
![image](https://user-images.githubusercontent.com/29524461/79994600-d08ba500-84d3-11ea-92a7-709870832d0f.png)
- Information of Credential API
![image](https://user-images.githubusercontent.com/29524461/79994741-f7e27200-84d3-11ea-95ab-7908360464f2.png)

- Messaging URL information 
![image](https://user-images.githubusercontent.com/29524461/79994288-5eb35b80-84d3-11ea-950b-42381d6669e3.png)
Or may be i am missing something in the response.

Reply:
I updated the iot Project type with a list of more generic `IotEndpoint` type, and added the device registry API details there. Let me know if that works.

> * Messaging URL information
>   ![image](https://user-images.githubusercontent.com/29524461/79994288-5eb35b80-84d3-11ea-950b-42381d6669e3.png)
>   Or may be i am missing something in the response.

I'll investigate this one  a bit more. :)